test/spawn_spec.js

```diff
@@ -5,19 +5,26 @@ const launch = require("../").launch;
 const kernelspecs = require("kernelspecs");
 
 describe("launch", () => {
-  it("spawns a kernel", () => {
-    return kernelspecs
+  let spawnResult;
+  it("spawns a kernel", done => {
+    kernelspecs
       .findAll()
       .then(kernels => {
         const kernel = kernels.python2 || kernels.python3;
         return launch(kernel.name);
       })
       .then(c => {
+        spawnResult = c;
         expect(c).to.not.be.null;
         expect(c.spawn).to.not.be.null;
-
+        expect(fs.existsSync(c.connectionFile)).to.be.true;
         c.spawn.kill();
-        fs.unlinkSync(c.connectionFile);
+        done();
       });
   });
+
+  it("cleans up connection files", () => {
+    const { connectionFile } = spawnResult;
+    expect(fs.existsSync(connectionFile)).to.be.false;
```

Comment:
Love it.

Reply:
I got travis to run, but this particular test will fail randomly if the file hasn't been deleted yet. This never happened when running locally. 

I could add a ~100 ms `setTimeout` to make sure there is enough time. I know it's janky, but it would still test the functionality properly. What do you think?

Reply:
I think that's fine since we don't have a callback for the file being deleted.

Reply:
Tests pass!